Marlin/Marlin_main.cpp

```diff
@@ -7233,6 +7233,16 @@ inline void gcode_M503() {
     if (code_seen('Z')) {
       float value = code_value_axis_units(Z_AXIS);
       if (Z_PROBE_OFFSET_RANGE_MIN <= value && value <= Z_PROBE_OFFSET_RANGE_MAX) {
+        #if ENABLED(UPDATE_GRID_WITH_PROBE_HEIGHT)
```

Comment:
maybe just rename UPDATE_GRID_WITH_PROBE_HEIGHT to AUTO_BED_LEVELING_BILINEAR ?
it seems like bug or omission than an option
if option enabled it must work in all cases properly